documents/privacy.md

```diff
@@ -0,0 +1,301 @@
+
+### Overall level of protection
+
+Since the space of locations and timestamps for a given Health Authority is relatively small, it has low entropy, and the overall level of protection we can provide against brute force attacks is limited.
+
+The following table shows the costs of various hashing operations to our systems, and the costs of various attacks, on the assumption that:
+
+-   we tune the cost of our hash to take approximately 1 second, on a typical smartphone
+-   the servers that encode hashes for Safe Places are approximately 10x more powerful
+-   the servers used by an attacker are approx 100x as powerful as the smartphone.
+-   servers of such power could be rented in the cloud (e.g. AWS) for 10c/hour
+
+(we are actively working to validate these assumptions)
+
+| **Activity**                                             | **Smartphone CPU seconds (SCS)** | **Estimated CPU time**         |
+| -------------------------------------------------------- | -------------------------------- | ------------------------------ |
+| Smartphone hashes 1 location                             | 4.6                              | 4.6 seconds                    |
+| Smartphone CPU usage/day                                 | 1.3k                             | 22 mins                        |
+| Safe Places publish (100 cases)                          | 100k                             | 3 hours                        |
+| Safe Places CPU usage/day                                | 200k                             | 6 hours                        |
+| Attack on single geohash, 1 day                          | 288                              | 3 seconds                      |
+| Attack on 1 sq km, 1 day                                 | 400k                             | 1 hour 6 mins (est. cost 12 c) |
+| Attack on 10km x 10km, 14 days                           | 560M                             | 2 months (est. cost $144.00)   |
+| Attack on 3000 sq km, 14 days (e.g. a typical US County) | 16.8B                            | 5 years (est. cost $4,560)     |
+
+
+
+Relating this to our privacy goals articulated above
+
+-   There is no significant risk to undiagnosed users (whose data is never shared), or diagnosed users (since the data they have
+    provided, even if it were decrypted, is anonymized and redacted).
+    
+-   There is, unfortunately, a vulnerability for venues where diagnosed users have been - particularly in the case where an attacker has in mind a small area and a narrow time window, which they wish to evaluate for points of concern.
+
+However, while such attacks are possible, it nevertheless will take significant technical skill and effort to conduct such an attack, and the rewards are fairly minimal. Our expectation is that this will lead to few such attacks will be undertaken, and that this will be a negligible channel for disclosure of information about which venues were attended by people later diagnoses with COVID.
```

Comment:
Our expectation is that this will lead to few such attacks **will be** undertaken should be Our expectation is that this will lead to few such attacks **being** undertaken
which venues were attended by people later **diagnoses** with COVID.  should be which venues were attended by people later **diagnosed** with COVID.